G-Research.OpenTelemetry.Processor.Partial/PartialActivityProcessor.cs

```diff
@@ -23,7 +24,7 @@ public class PartialActivityProcessor : BaseProcessor<Activity>
 
     private readonly BaseExporter<LogRecord> logExporter;
     private readonly BaseProcessor<LogRecord> logProcessor;
-    private ILoggerFactory loggerFactory;
+    private ILoggerFactory loggerFactory = null!;
```

Comment:
If the `loggerFactory` is never created (e.g. no span/activity is ever created), then I think that the call to `Dispose` will fail because it's `null` - could you please guard against this in the `Dispose` method?

Alternatively, you can copy my code that defined a `Lazy<ILoggerFactory>`.